bgpd/bgp_route.c

```diff
@@ -5800,17 +5800,27 @@ static void bgp_aggregate_install(struct bgp *bgp, afi_t afi, safi_t safi,
 			aggregate, atomic_aggregate, p);
 
 		if (!attr) {
+			char buf1[INET6_ADDRSTRLEN];
+			inet_ntop(p->family, &p->u.prefix, buf1, INET6_ADDRSTRLEN);
 			bgp_aggregate_delete(bgp, p, afi, safi, aggregate);
+			zlog_info("%s(%s): null attribute", __FUNCTION__, buf1);
 			return;
 		}
 
-		new = info_make(ZEBRA_ROUTE_BGP, BGP_ROUTE_AGGREGATE, 0,
-				bgp->peer_self, attr, rn);
+		/* Do not install the aggregate route if BGP is in the
+		 * process of termination.
+		 */
+		if (bgp->peer_self &&
+					!bgp_flag_check(bgp, BGP_FLAG_DELETE_IN_PROGRESS)) {
```

Comment:
CHECK_FLAG(bgp->flags, BGP_FLAG_DELETE_IN_PROGRESS)

Reply:
both are same. In BGP, we already have this function. So, I think it is fine to use the BGP function for readability.
int bgp_flag_check(struct bgp *bgp, int flag)
{
	return CHECK_FLAG(bgp->flags, flag);
}

Reply:
No, we don't have it already. It was before. Please rebase to latest master.

Reply:
done